llmfoundry/data/finetuning/tasks.py

```diff
@@ -152,65 +157,100 @@ def wrapper(func: Callable) -> Callable:
 
         return wrapper
 
-    def build(self, cfg: DictConfig, tokenizer: Tokenizer):
-        dataset_name = cfg.name
-        split = cfg.split
-        kwargs = cfg.get('kwargs', {})
+    def print_registered_tasks(self):
+        tasks = sorted(self._task_preprocessing_registry.keys())
+        print('\n'.join(tasks))
+
+    def get_preprocessing_fn_from_str(self,
+                                      preprocessor: Optional[str],
+                                      dataset_name: Optional[str] = None,
+                                      verbose: bool = False):
+        if preprocessor is None:
```

Comment:
```suggestion
    """
    Get a preprocessing function from a string, either a registered function or an import path.

    Args:
        preprocessor (Optional[str]): The name of the preprocessing function, or an import path.
        dataset_name (Optional[str]): The dataset name to look up in the registry.
        verbose (bool): Whether to print verbose messages or not.

    Returns:
        Callable: The preprocessing function or None if not found.

    Raises:
        ValueError: If the preprocessing function import from the provided string fails.
    """
        if preprocessor is None:
```

trying to add docstrings to the important ones, since I think this code will get heavy usage